pkg/resources/stream_on_table_acceptance_test.go

```diff
@@ -0,0 +1,564 @@
+						HasAppendOnlyString(r.BooleanFalse).
+						HasTableString(table.ID().FullyQualifiedName()),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_StreamOnTable_CopyGrants(t *testing.T) {
+	acc.TestAccPreCheck(t)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_stream_on_table.test"
+
+	var createdOn string
+
+	table, cleanupTable := acc.TestClient().Table.CreateWithChangeTracking(t)
+	t.Cleanup(cleanupTable)
+	model := model.StreamOnTable("test", id.DatabaseName(), id.Name(), id.SchemaName(), table.ID().FullyQualifiedName())
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnTable),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(true)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						createdOn = value
+						return nil
+					})),
+				),
+			},
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(false)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						if value != createdOn {
+							return fmt.Errorf("view was recreated")
+						}
+						return nil
+					})),
+				),
+			},
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(true)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						if value != createdOn {
+							return fmt.Errorf("view was recreated")
+						}
+						return nil
+					})),
+				),
+			},
+		},
+	})
+}
+
+// There is no way to check at/before fields in show and describe. That's why in we try creating with these values, but do not assert them.
+func TestAcc_StreamOnTable_At(t *testing.T) {
+	acc.TestAccPreCheck(t)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceId := helpers.EncodeResourceIdentifier(id)
+	resourceName := "snowflake_stream_on_table.test"
+
+	table, cleanupTable := acc.TestClient().Table.CreateWithChangeTracking(t)
+	t.Cleanup(cleanupTable)
+	acc.TestClient().Table.InsertInt(t, table.ID())
+
+	lastQueryId := acc.TestClient().Context.LastQueryId(t)
+
+	model := model.StreamOnTable("test", id.DatabaseName(), id.Name(), id.SchemaName(), table.ID().FullyQualifiedName()).
+		WithComment("foo").WithAppendOnly(r.BooleanTrue).WithShowInitialRows(r.BooleanTrue).WithCopyGrants(true)
```

Comment:
Just a stylistic comment, I think it's more readable when every WithX has its own line

Reply:
Done.